crates/server/src/main.rs

```diff
@@ -2,18 +2,53 @@ mod actions;
 mod catalog;
 mod server;
 
+use std::net::IpAddr;
+
 use arrow_flight::flight_service_server::FlightServiceServer;
+use clap::AppSettings::ArgRequiredElseHelp;
+use clap::{
+    AppSettings::{DeriveDisplayOrder, DisableHelpSubcommand, InferLongArgs},
+    Parser,
+};
 use log::info;
 use tonic::transport::Server;
 
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
     env_logger::init();
-    let addr = "[::1]:50051".parse()?;
+
+    let CliOpts { host, port } = CliOpts::new();
+
+    let addr = std::net::SocketAddr::new(host, port);
     let service = crate::server::FlightServiceImpl::new();
     let service = FlightServiceServer::new(service);
     info!("Starting server at {addr}");
     Server::builder().add_service(service).serve(addr).await?;
 
     Ok(())
 }
+
+#[derive(Debug, Parser)]
+#[clap(
+    version,
+    about = "Graph Arrow Server",
+    global_setting = ArgRequiredElseHelp,
+    global_setting = DeriveDisplayOrder,
+    global_setting = DisableHelpSubcommand,
+    global_setting = InferLongArgs,
```

Comment:
That looks familiar ^^

Reply:
https://github.com/s1ck/graph/pull/27/commits/9a51726e3c15684e4431fa95882b9a72173c13d0